applications/OptimizationApplication/tests/control/shape/test_vm_shape_control.py

```diff
@@ -0,0 +1,195 @@
+                "filter_settings": {
+                    "type" : "surface_explicit_with_mesh_motion",
+                    "radius": 0.5
+                },
+                "fixed_model_parts" : {
+                    "solid_explicit.fixed" : [true,true,true]
+                }
+            }""")
+            cls.explicit_solid_shape_control = VertexMorphingShapeControl("solid_explicit", cls.model, solid_parameters, cls.optimization_problem)
+            cls.optimization_problem.AddComponent(cls.explicit_solid_shape_control)
+            ReadModelPart("solid",cls.solid_explicit)
+            cls.explicit_solid_shape_control.Initialize()
+
+            cls.solid_implicit = cls.model.CreateModelPart("solid_implicit")
+            cls.solid_implicit.ProcessInfo[Kratos.DOMAIN_SIZE] = 3
+            solid_parameters = Kratos.Parameters("""{
+                "controlled_model_part_names": ["solid_implicit.design"],
+                "filter_settings": {
+                    "type" : "surface_implicit_with_mesh_motion",
+                    "radius": 0.5
+                },
+                "fixed_model_parts" : {
+                    "solid_implicit.fixed" : [true,true,true]
+                }
+            }""")
+
+            cls.implicit_solid_shape_control = VertexMorphingShapeControl("solid_implicit", cls.model, solid_parameters, cls.optimization_problem)
+            cls.optimization_problem.AddComponent(cls.implicit_solid_shape_control)
+            ReadModelPart("solid",cls.solid_implicit)
+            cls.implicit_solid_shape_control.Initialize()
+
+
+            cls.solid_bulk_surf_implicit = cls.model.CreateModelPart("solid_bulk_surf_implicit")
+            cls.solid_bulk_surf_implicit.ProcessInfo[Kratos.DOMAIN_SIZE] = 3
+            solid_parameters = Kratos.Parameters("""{
+                "controlled_model_part_names": ["solid_bulk_surf_implicit.design"],
+                "filter_settings": {
+                    "type" : "bulk_surface_implicit",
+                    "radius": 0.5
+                },
+                "fixed_model_parts" : {
+                    "solid_bulk_surf_implicit.fixed" : [true,true,true]
+                }
+            }""")
+
+            cls.implicit_bulk_surf_shape_control = VertexMorphingShapeControl("solid_bulk_surf_implicit", cls.model, solid_parameters, cls.optimization_problem)
+            cls.optimization_problem.AddComponent(cls.implicit_bulk_surf_shape_control)
+            ReadModelPart("solid",cls.solid_bulk_surf_implicit)
+            cls.implicit_bulk_surf_shape_control.Initialize()
+
+    @classmethod
+    def tearDownClass(cls):
+        with kratos_unittest.WorkFolderScope(".", __file__):
+            DeleteFileIfExisting("shell.time")
+
+    def test_GetControlField(self):
+        implicit_control_field = self.implicit_shape_control.GetControlField()
+        self.assertAlmostEqual(KratosOA.ExpressionUtils.NormInf(implicit_control_field), 1.0903407171261381, 10)
```

Comment:
It would be better to compare assertVectorAlmostEqual. It may happend that the max value is the save while the rest changes.